File: ws/bonilla/U1/WS10Constructors/ChickenFarmSimulatorJs/ec.espe.edu.model/Chicken.js
export class Chicken  {
  constructor(id, name, color, age, molting, bornOnDate) {
    this._id = id;
    this._name = name;
    this._color = color;
    this._age = age;
    this._molting = molting;
    this._bornOnDate = bornOnDate;
  }

  toString() {
    return `Chicken{id=${this.getId()}, name=${this.getName()}, color=${this.getColor()}, age=${this.getAge()}, molting=${this.isMolting()}, bornOnDate=${this.getBornOnDate()}}`;
  }

  duStuff() {
    console.log("Chicken is ...");
    this.#cluck();
    this.#wander();
  }

  // Métodos privados (disponibles en Node.js modernos o navegadores actualizados)
  #cluck() {
    console.log("chicken name --> " + this.getName());
    console.log("chicken date --> " + this.getBornOnDate());
    console.log(`The chicken ${this.getId()} is clucking`);
  }

  #wander() {
    console.log(`The chicken ${this.getId()} is wandering`);
  }

  #eat() {
    console.log(`The chicken ${this.getId()} is eating`);
  }

  #drink() {
    console.log(`The chicken ${this.getId()} is drinking`);
  }

  // Getters y setters
  getId() {
    return this._id;
  }

  setId(id) {
    this._id = id;
  }

  getName() {
    return this._name;
  }

  setName(name) {
    this._name = name;
  }

  getColor() {
    return this._color;
  }

  setColor(color) {
    this._color = color;
  }

  getAge() {
    return this._age;
  }

  setAge(age) {
    this._age = age;
  }

  isMolting() {
    return this._molting;
  }

  setMolting(molting) {
    this._molting = molting;
  }

  getBornOnDate() {
    return this._bornOnDate;
  }

  setBornOnDate(bornOnDate) {
    this._bornOnDate = bornOnDate;
  }
}

File: ws/maisincho/u1/WS10Constructors/ChickenSimulator.py/chicken.py
from datetime import datetime

class Chicken:
    def __init__(self, id=None, name=None, color=None, age=None, molting=None, born_on_date=None):
        self.id = id
        self.name = name
        self.color = color
        self.age = age
        self.molting = molting
        self.born_on_date = born_on_date

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_color(self):
        return self.color

    def set_color(self, color):
        self.color = color

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

    def is_molting(self):
        return self.molting

    def set_molting(self, molting):
        self.molting = molting

    def get_born_on_date(self):
        return self.born_on_date

    def set_born_on_date(self, born_on_date):
        self.born_on_date = born_on_date

    def __str__(self):
        return (f"Chicken{{id={self.get_id()}, name={self.get_name()}, color={self.get_color()}, "
                f"age={self.get_age()}, molting={self.is_molting()}, bornOnDate={self.get_born_on_date()}}}")

    def do_stuff(self):
        print("Chicken is -->")
        self.cluck()
        self.wander()

    def cluck(self):
        print(f"Chicken name --> {self.get_name()}")
        print(f"Chicken date --> {self.get_born_on_date()}")
        print(f"Chicken --> {self.get_id()} is clucking")

    def wander(self):
        print(f"Chicken --> {self.get_id()} is wandering")

    def eat(self):
        print(f"Chicken --> {self.get_id()} is eating")

    def drink(self):
        print(f"Chicken --> {self.get_id()} is drinking")
